### PR TITLE
Sandbox server & `@inline` directive

### PR DESCRIPTION
**Pull request by gdotdesign**
This PR adds the `sandbox-server` command to start a server for compiling and formatting files for the sandbox (https://sandbox.mint-lang.com/) and for the new tutorial site (https://tutorial.mint-lang.com/introduction/basics).

It also adds the `@inline` directive which inlines a text file to the compiled JavaScript code.

### CODE REVIEW

File: src/type_checkers/directives/inline.cr [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  class TypeChecker
+    type_error AssetDirectiveExpectedFile
+
+    def check(node : Ast::Directives::Inline) : Checkable
+      raise AssetDirectiveExpectedFile, {
+        "path" => node.real_path.to_s,
+        "node" => node,
+      } unless node.exists?
+
+      # puts File.new(node.real_path).encoding
```
Comment (Sija) [inline src/type_checkers/directives/inline.cr:11]: Debug leftover?

File: src/sandbox_server.cr [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+      template =
+        TreeTemplate.new(formatter: TreeTemplate::PrettyFormatter) do |t|
+          t.doctype :html5
+
+          t.html do
+            t.head do
+              t.meta(charset: "utf-8")
+
+              # t.title json.application.title.to_s
```
Comment (Sija) [inline src/sandbox_server.cr:104]: ditto

File: src/commands/sandbox_server.cr [hunk 3]
```diff
@@ -0,0 +1,24 @@
+module Mint
+  class Cli < Admiral::Command
+    class SandboxServer < Admiral::Command
+      define_help description: "Server for compiling sandbox applications"
+
+      define_flag host : String,
+        description: "Change the host the server binds to. (Default: 0.0.0.1)",
+        default: ENV["HOST"]? || "0.0.0.1",
```
Comment (Sija) [inline src/commands/sandbox_server.cr:8]: I'd use either `127.0.0.1` or `0.0.0.0`.

**Review by Sija — CHANGES_REQUESTED** [review]

File: src/commands/sandbox_server.cr (hunk 3 above)
Comment (gdotdesign) [reply src/commands/sandbox_server.cr:8]: Yes, that's supposed to be `0.0.0.0`

File: spec/compilers_spec.cr [hunk 4]
```diff
@@ -14,7 +14,12 @@ Dir
       ast.class.should eq(Mint::Ast)
 
       # Compare results
-      result = Mint::Compiler.compile_bare(Mint::TypeChecker.check(ast))
+      result = Mint::Compiler.compile_bare(Mint::TypeChecker.check(ast), {
+        optimize:   false,
+        relative:   false,
+        build:      true,
+        css_prefix: nil,
```
Comment (Sija) [inline spec/compilers_spec.cr:21]:
```suggestion
        css_prefix: nil,
        optimize:   false,
        relative:   false,
        build:      true,
```

**Review by Sija — APPROVED** [review]
LGTM 🚀

**Pull request merged**